specs/0109-infobar-component.md

```diff
@@ -0,0 +1,71 @@
+* Feature Name: Infobar Component
+* Created: 2021-02-25
+* Associated Bugs: [bug 1694911](https://bugzilla.mozilla.org/show_bug.cgi?id=1694911)
+
+# Summary
+
+Create a reusable component to display infobars to users.
+
+# Motivation
+
+We need a flexible way to notify users without writing ad-hoc code for every situation:
+* Infobars should be manageable by admins without intervention from developers.
+* Unlike notifications, infobars are more visible and require interaction from users to get dismissed.
+
+Example use cases:
+* We introduce a brand new feature, like string comments, and we want to make sure that users are aware of it. The infobar will include a short message and a link to documentation.
+* We want to warn users of an upcoming maintenance window for Pontoon.
+
+# Feature explanation
+
+When a logged in user opens Pontoon, we check against a list of infobars. If the user matches the necessary conditions, we display the infobar.
+
+Information about each infobar is stored in a table with the following fields:
+* Identifier: short text identifier, e.g. `new_terms_2020`.
+* Start time (datetime): when we should start displaying the infobar to users.
+* End time (datetime): when we should stop displaying the infobar to users.
+* Creation date (date): stores when the infobar was created.
+* Cohort: if defined, we only display the infobar for users part of a specific experiment or cohort (A/B testing).
```

Comment:
Give this an `(optional)` to align with the other fields?

Also, should we have a modifier for things that we can't implement right now, as they're blocked by other things to be specified?

Reply:
I think it's OK to define a field in the spec for future use, without explicitly call it out? The alternative would be to drop it, and add it later if/when we decide to implement something that uses it.